File: 0189/sol2.cpp
#include <bits/stdc++.h>
using namespace std;

void printVector(vector<int> &nums) {
    for (int num : nums) {
        cout << num << ' ';
    }
    cout << endl;
}

class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        if (k % nums.size() == 0) return;
        int n = nums.size();
        k %= n;
        int count = 0, start = 0, cur = 0, val = nums.at(cur);
        while (count < n) {
            cur += k;
            cur %= n;
            swap(val, nums.at(cur));
            if (cur == start) {
                cur = ++start;
                val = nums.at(cur);}
            count++;
        }
    }
};

int main() {
    vector<int> nums = {1,2};
    int k = 2;
    Solution sol;
    sol.rotate(nums, k);
    printVector(nums);

    return 0;
}

File: 0189/sol2.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        
        n = len(nums)
        k %= n

        i = n - k
        nums[:] = nums[i:] + nums[:i]

# test cases
sol = Solution()
nums = [-1,-100,3,99]
k = 2
sol.rotate(nums, k)

print(
    nums
)